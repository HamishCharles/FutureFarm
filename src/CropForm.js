import React, { Component } from 'react';
import {calcYeildValue}  from './businessLogic';

class CropForm extends Component {
 
  render() {
    const {field, changeCrop, fieldCrop, crops, fieldYeild} = this.props;
    console.log(this.props)
    return (
      <form><h4>{field.name} &emsp; <span className='yield'>{ fieldYeild ? 'Yield Â£' + Math.round(fieldYeild) : ''}</span></h4> 
        <select onChange={changeCrop} value={fieldCrop}>
          <option key="" value="">No crop selected</option>
          {crops.map((crop, idx) => <option key={crop.name} value={idx}>
            {crop.name} (&pound;{Math.round(calcYeildValue(crop.expected_yield, field.hectares, crop.disease_risk_factor, field.disease_susceptibility, crop.price_per_tonne))})
          </option>)}
        </select>
      </form>
    )
  }
}

export default CropForm;
