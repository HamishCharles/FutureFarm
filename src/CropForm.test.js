import React from 'react';
// import ReactDOM from 'react-dom';
import CropForm from './CropForm';
import renderer from 'react-test-renderer';


test('Renders crop selection form correctly', () => {
  const component = renderer.create(
    <CropForm field={{name: 'test field'}} changeCrop={function () {}} fieldCrop={1} crops={[{name: 'Winter rice'},{name:'Summer oats'}]} fieldYeild={50}></CropForm>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});