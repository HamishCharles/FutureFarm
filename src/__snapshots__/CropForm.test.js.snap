// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders crop selection form correctly 1`] = `
<form>
  <h4>
    test field
       
    <span
      className="yield"
    >
      Yield £50
    </span>
  </h4>
  <select
    onChange={[Function]}
    value={1}
  >
    <option
      value=""
    >
      No crop selected
    </option>
    <option
      value={0}
    >
      Winter rice
       (£
      NaN
      )
    </option>
    <option
      value={1}
    >
      Summer oats
       (£
      NaN
      )
    </option>
  </select>
</form>
`;
